units/src/amount.rs

```diff
@@ -1158,7 +1158,13 @@ impl ops::DivAssign<u64> for Amount {
 impl FromStr for Amount {
     type Err = ParseError;
 
-    fn from_str(s: &str) -> Result<Self, Self::Err> { Amount::from_str_with_denomination(s) }
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if !s.chars().any(char::is_alphabetic) && s.parse::<f64>() == Ok(0.0) && !s.starts_with('-') {
```

Comment:
I can not easily reason about the statement `!s.chars().any(char::is_alphabetic)`

The unit tests seem to implement what was discussed in the issue and the following code passes the tests and is more simple

```rust
        // `Amount` is an unsigned type so we do not allow `-`.
        if s.parse::<f64>() == Ok(0.0) && !s.starts_with('-') {
            Ok(Amount::ZERO)
        } else {
            Amount::from_str_with_denomination(s)
        }
```

Reply:
Yeah, that condition is useless. But we should also have a test that the smallest non-zero value (as restricted by max len) does not parse.

Reply:
@tcharding thanks.  The condition was to guard against anything that did have a denomination from being parsed as float, although it's not needed anymore.